include/deal.II/base/thread_management.h

```diff
@@ -1025,7 +1032,8 @@ namespace Threads
      * Get the return value of the function of the thread. Since this is only
      * available once the thread finishes, this implicitly also calls join().
      */
-    RT return_value ()
+    typename internal::return_value<RT>::reference_type
+    return_value ()
```

Comment:
And this (and the other place below for class `Task`) is the other key place: instead of returning the value returned by the thread/task function by value, we need to return a reference to it so that accessing functions don't have to move it away and can ask for it more than once.